packages/relay/tests/lib/eth/eth_getLogs.spec.ts

```diff
@@ -439,27 +478,29 @@ describe('@ethGetLogs using MirrorNode', async function () {
     expectLogData1(result[0]);
   });
 
-  it('when block range is too large', async function () {
-    const fromBlock = {
-      ...DEFAULT_BLOCK,
-      number: 1,
-    };
-    const toBlock = {
-      ...DEFAULT_BLOCK,
-      number: 1003,
-    };
-
-    const blockBeyondMaximumRange = {
-      ...DEFAULT_BLOCK,
-      number: 1007,
-    };
-
-    restMock.onGet(BLOCKS_LIMIT_ORDER_URL).reply(200, { blocks: [blockBeyondMaximumRange] });
-    restMock.onGet('blocks/1').reply(200, fromBlock);
-    restMock.onGet('blocks/1003').reply(200, toBlock);
-
-    await ethGetLogsFailing(ethImpl, [null, '0x1', '0x3eb', null, null], (error) => {
-      expect(error.message).to.equal('Exceeded maximum block range: 1000');
+  [null, [], [CONTRACT_ADDRESS_1, CONTRACT_ADDRESS_2]].forEach((address) => {
+    it(`when block range is too large with address \`${JSON.stringify(address)}\``, async function () {
```

Comment:
I see that we are trying to create many tests out of a collection, but the same could be achieved as part of the same test. and would be cleaner in my opinion.

Also the name of the test should be more descriptive and say what is expecting to happen.

like: `should fail when block range is too large for various addresses`

```
it('should fail when block range is too large for various addresses', async function () {
  const addresses = [null, [], [CONTRACT_ADDRESS_1, CONTRACT_ADDRESS_2]];
  for (const address of addresses) {
    const fromBlock = {
      ...DEFAULT_BLOCK,
      number: 1,
    };
    const toBlock = {
      ...DEFAULT_BLOCK,
      number: 1003,
    };

    const blockBeyondMaximumRange = {
      ...DEFAULT_BLOCK,
      number: 1007,
    };

    // Mocking the HTTP responses for the blocks limit order URL and for specific block numbers
    restMock.onGet(BLOCKS_LIMIT_ORDER_URL).reply(200, { blocks: [blockBeyondMaximumRange] });
    restMock.onGet('blocks/1').reply(200, fromBlock);
    restMock.onGet('blocks/1003').reply(200, toBlock);

    // Testing the failure scenario for the ethGetLogs function with various addresses
    await ethGetLogsFailing(ethImpl, [null, '0x1', '0x3eb', address, null], (error) => {
      expect(error.message).to.equal('Exceeded maximum block range: 1000', `Failed for address: ${JSON.stringify(address)}`);
    });
  }
});

```

Reply:
As discussed with Fready, we left the dynamic test cases, but improve the test title to match `it should` format.